lib/SILOptimizer/Mandatory/TFUtilities.cpp

```diff
@@ -257,6 +260,16 @@ SILFunction *tf::findSILFunctionForRequiredProtocolMember(
   return nullptr;
 }
 
+SubstitutionMap tf::getSingleSubstitutionMapForElementType(Type ty,
+                                                           ASTContext &ctx) {
+  auto builder = GenericSignatureBuilder(ctx);
```

Comment:
Does it really make sense to create a new builder every time? Why don't you store this trivial signature somewhere?

Reply:
Added a FIXME comment -- favoring code simplicity for now, and will address it if there is non-trivial runtime overhead.